inc/namespace.php

```diff
@@ -246,6 +262,130 @@ function noop_wp_query_found_rows_on_failed_ep_request( string $sql, WP_Query $q
 	return '';
 }
 
+/**
+ * Add default initial options and settings on install.
+ *
+ * @return void
+ */
+function on_wp_install() {
+	// This option is used to determine the index name for backwards compat.
+	set_index_version( 1 );
+}
+
+/**
+ * When an existing index is deleted set the index version so
+ *
+ * @param array $query The Elasticsearch query.
+ * @param string|null $type The remote request type.
+ * @return void
+ */
+function on_delete_index( $query, ?string $type ) {
+	if ( $type !== 'delete_index' ) {
+		return;
+	}
+	// Set the version if not already present.
+	if ( ! get_index_version() ) {
+		set_index_version( 1 );
+	}
+}
+
+/**
+ * Set the index version for the current site.
+ *
+ * @param integer $version The version number.
+ * @return void
+ */
+function set_index_version( int $version ) {
+	update_option( 'altis_search_index_version', $version );
+}
+
+/**
+ * Get the index version for the current site.
+ *
+ * @return int|null
+ */
+function get_index_version() : ?int {
+	return get_option( 'altis_search_index_version', null );
```

Comment:
Maybe `?? 1` if we implement suggestion above so this always returns `int`?

Reply:
No, we need something equivalent to the option not being set at all for this to work

Reply:
Yeah, I mean `?? 1` here *and* the actual set version as `3` (so `1` is equivalent to unset).

Reply:
Seems a bit weird no? Null seems more meaningful here to me but I think otherwise the fallback value could be 2 if we’re representing the elastic press version

Reply:
2 seems fine as well yeah. Provided it's installed, the indexes always have a version, it's just whether it's explicit or not, so better to fall back I think.

Reply:
Yep, makes sense. I've updated it now so the default value is 2 for EP v2.